### PR TITLE
Wishlist page and functionality

### PR DESCRIPTION
**Pull request by sushil-mod**
# Added Wishlist page and Functionality 
how to check :- login as a guest -> go to product page by clicking shop now -> then click on heart badge on product card then from nav also same 

## What to check 

- Wishlist :-  src/pages/wishlist/Wishlist.jsx
- sevices :-  src/service calls/sevices.jsx
-  ProductCard  :-  src/components/card/ProductCard.jsx

### CODE REVIEW

File: src/service calls/sevices.jsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        console.error(error);
+    }
+}
+
+
+const addToWishlist = async (product,{encodedToken,authDispatch})=>{
+    
+    try {
+        const { data } = await axios.post("/api/user/wishlist",{product},{headers:{ authorization : encodedToken }})
+        console.log("form atw",data)
```
Comment (rohit8686) [inline src/service calls/sevices.jsx:21]: remove console statements

File: src/service calls/sevices.jsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    } catch (error) {
+        console.error(error)
+    }
+}
+
+const removeFromWishlist = async (productId,{encodedToken,authDispatch})=>{
+    
+    try {
+        const { data } = await axios.delete( `/api/user/wishlist/${productId}`,{headers:{ authorization : encodedToken }})
+        console.log("form atw",data)
```
Comment (rohit8686) [inline src/service calls/sevices.jsx:32]: here also

**Review by rohit8686 — APPROVED** [review]
clean code and looks good to merge

File: src/service calls/sevices.jsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import axios from 'axios';
+import React from 'react';
+
+
+const addToCart = async (product,{encodedToken,authDispatch,navigator })=>{
+    try {
+        const response = await axios.post("/api/user/cart",{product},{headers:{ authorization : encodedToken }})
```
Comment (JantuDeb) [inline src/service calls/sevices.jsx:7]: you can destructure `data` from response `const  {data} = await ...`

File: src/service calls/sevices.jsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    } catch (error) {
+        console.error(error)
+    }
+}
+
+const productQtyHandler = async( productId , type ,{encodedToken,authDispatch } ) => {
+   
+    try {
+     const response = await axios.post(`/api/user/cart/${productId}`,{action:{type:`${type}`}},{headers:{authorization : encodedToken }})   
+        authDispatch({ type:"QUANTITY" , payload : response.data.cart})
```
Comment (JantuDeb) [inline src/service calls/sevices.jsx:43]: same here

**Review by JantuDeb — APPROVED** [review]
overall codes looks good

File: src/service calls/sevices.jsx (hunk 3 above)
Comment (sushil-mod) [reply src/service calls/sevices.jsx:7]: i always forgot this , thanks for pointing ou again

File: src/service calls/sevices.jsx (hunk 4 above)
Comment (sushil-mod) [reply src/service calls/sevices.jsx:43]: yes

File: src/service calls/sevices.jsx (hunk 2 above)
Comment (sushil-mod) [reply src/service calls/sevices.jsx:32]: ok ,done

File: src/service calls/sevices.jsx (hunk 1 above)
Comment (sushil-mod) [reply src/service calls/sevices.jsx:21]: yes

**Pull request merged**